File: models/event.go
package models

import (
	"database/sql"
	"time"
)

type Event struct {
	ID             int            `json:"id"`
	Title          string         `json:"title"`
	Description    sql.NullString `json:"-"`           // ✅ SANS pointeur
	DescriptionStr string         `json:"description"` // ✅ Pour le JSON
	CreatedBy      int            `json:"created_by"`
	CreatorName    string         `json:"creator_name"`
	CreatedAt      time.Time      `json:"created_at"`
	Latitude       float64        `json:"latitude"`
	Longitude      float64        `json:"longitude"`
	Address        string         `json:"address"`
	StartDatetime  time.Time      `json:"start_datetime"`
	EndDatetime    *time.Time     `json:"end_datetime,omitempty"`
	Participants   int            `json:"participants"`
	UserJoined     bool           `json:"user_joined"`
	FormattedStart string         `json:"-"` // ✅ Pour le template HTML
}

File: mobile/lib/data/models/event.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:latlong2/latlong.dart';

part 'event.freezed.dart';
part 'event.g.dart';

@Freezed()
class Event with _$Event {
  @JsonSerializable(fieldRename: FieldRename.snake)
  const factory Event({
    required String id,
    required String title,
    String? description,
    required String createdBy,
    String? creatorName,
    required double latitude,
    required double longitude,
    String? address,
    required DateTime startDatetime,
    DateTime? endDatetime,
    @Default(0) int participants,
    @Default(false) bool userJoined,
    required DateTime createdAt,
  }) = _Event;

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);
}

extension EventX on Event {
  LatLng get latLng => LatLng(latitude, longitude);
}
